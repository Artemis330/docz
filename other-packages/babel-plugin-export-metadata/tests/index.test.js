const { transformSync } = require('@babel/core')
const fs = require('fs')
const path = require('path')
const plugin = require('../src')

const fixture = path.resolve('./tests/fixtures/example.js')
const code = fs.readFileSync(fixture).toString()

it('works', () => {
  const result = transformSync(code, {
    plugins: [plugin],
    filename: fixture,
  })

  expect(result.code).toMatchSnapshot()
})
