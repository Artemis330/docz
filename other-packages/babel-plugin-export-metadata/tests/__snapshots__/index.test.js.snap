// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`works 1`] = `
"const foo = () => 'foo';

export const bar = () => 'bar';

if (bar === Object(bar)) {
  Object.defineProperty(bar, '__filemeta', {
    writable: false,
    value: {
      name: \\"bar\\",
      filename: \\"/Volumes/Projects/dev/docz/docz/other-packages/babel-plugin-export-metadata/tests/fixtures/example.js\\"
    }
  });
}

if (bar === Object(bar)) {
  Object.defineProperty(bar, '__filemeta', {
    writable: false,
    value: {
      name: \\"bar\\",
      filename: \\"/Volumes/Projects/dev/docz/docz/other-packages/babel-plugin-export-metadata/tests/fixtures/example.js\\"
    }
  });
}

const obj = {
  foo: () => 'foo'
  /**
   * Some description
   */

};

function get(object, path) {
  return object[path];
}

export function getOther(object, path) {
  return object[path];
}

if (getOther === Object(getOther)) {
  Object.defineProperty(getOther, '__filemeta', {
    writable: false,
    value: {
      name: \\"getOther\\",
      filename: \\"/Volumes/Projects/dev/docz/docz/other-packages/babel-plugin-export-metadata/tests/fixtures/example.js\\"
    }
  });
}

const obj2 = {
  get: () => null
  /**
   * Some description
   */

};

class Abc {
  /**
   * Some description
   */
  method() {
    return null;
  }

}

export const component = styled.div\`
  background: red;
\`;

if (component === Object(component)) {
  Object.defineProperty(component, '__filemeta', {
    writable: false,
    value: {
      name: \\"component\\",
      filename: \\"/Volumes/Projects/dev/docz/docz/other-packages/babel-plugin-export-metadata/tests/fixtures/example.js\\"
    }
  });
}

export class Abcd {
  /**
   * Some description
   */
  method() {
    return null;
  }

}

if (Abcd === Object(Abcd)) {
  Object.defineProperty(Abcd, '__filemeta', {
    writable: false,
    value: {
      name: \\"Abcd\\",
      filename: \\"/Volumes/Projects/dev/docz/docz/other-packages/babel-plugin-export-metadata/tests/fixtures/example.js\\"
    }
  });
}

const obj3 = {
  method: class Abcd {
    method() {
      return null;
    }

  }
};

const log = fn => fn(1);

log(id => console.log(id));
export default (() => 'other');"
`;
